src/Engine/GameAction.java

```diff
@@ -820,9 +820,14 @@ public GameEventQueue getEvents(MapMaster map)
         // Note that movePath being null is OK for ResupplyAction when it is being re-used.
         if( movePath != null )
         {
-          eventSequence.add(new MoveEvent(unitActor, movePath));
+          // If we should be blocked, don't resupply anything.
+          if( !Utils.enqueueMoveEvent(map, unitActor, movePath, eventSequence) )
+            isValid = false;
```

Comment:
If `enqueueMoveEvent()` fails, it doesn't mean the action is invalid; it means the unit got ambushed, so the unit didn't reach its destination.
Like, I think the logic here works, but the usage of `isValid` is no longer consistent with the other actions. I'd rather seethis expressed as `isValid &= (null != movePath);`, and then `if( Utils.enqueueMoveEvent() ) { /*resupply stuff */ }`

Reply:
My reasoning: it's no longer a valid *resupply* action, and the move's been queued already.

Also, `movePath == null` is a valid action, that's what gets pushed out on turn init.

Reply:
Ah, right. I forgot a null move path is valid for resupply. But I think you understand the intent behind what I wrote.
And an attack action that gets preempted is still a valid action, even though the attack itself won't occur. I'm not saying this code won't work, I'm just saying the treatment of `isValid` might be misleading to whoever looks at it in three years.

Reply:
I mean, I guess I could make a second boolean, but that seems needless.
I could return there, but that's got its own weirdnesses.
Mebbe just another comment, specifying the exceptional case?

Reply:
I mean... I might make a second Boolean if it were me, but I lean towards verbose code. I guess it doesn't matter all that much; a comment stating that the unit can still move despite setting `isValid` to false is probably sufficient. I know that's obvious to someone who knows how `enqueueMoveEvent()` works, but that's the kind of thing that won't be obvious if you haven't looked at it in a while.